packit/upstream.py

```diff
@@ -384,14 +384,21 @@ def create_archive(self, version: str = None) -> str:
             outputs = self.get_output_from_action(
                 action=ActionName.create_archive, env=env
             )
-            if not outputs:
-                raise PackitException("No output from create-archive action.")
+            if self.package_config.archive_name:
+                archive_path = Path(
+                    self._local_project.working_dir,
+                    self.package_config.archive_name.format(version=version),
+                )
+            else:
+                if not outputs:
+                    raise PackitException("No output from create-archive action.")
 
-            archive_path = self._get_archive_path_from_output(outputs)
-            if not archive_path:
+                archive_path = self._get_archive_path_from_output(outputs)
+            if not archive_path or not archive_path.exists():
                 raise PackitException(
                     "The create-archive action did not output a path to the generated archive. "
```

Comment:
needs updating, but you probably know that :) it could have print it but it may have been incorrect